dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESClient.java

```diff
@@ -312,59 +321,35 @@ private void restartNode(Builder externalSettings) {
         initNode(externalSettings);
     }
 
+    private String getServerAddress(final Server server) {
+        return  (UtilMethods.isSet(server.getHost()) && !server.getHost().equals("localhost"))
+            ? server.getHost()
+            : server.getIpAddress();
+    }
+
     /**
      *
-     * @param serverAPI
-     * @param currentServer
      * @param externalSettings
      * @throws DotDataException
      */
-    private void setUnicastHosts(final ServerAPI serverAPI,
-            Server currentServer, final Builder externalSettings) throws DotDataException {
-
-        String initData;
+    private void setUnicastHosts(final Builder externalSettings) throws DotDataException {
 
         final String bindAddr = externalSettings.get("transport.host");
-        final String transportTCPPort = externalSettings
-                .get(ServerPort.ES_TRANSPORT_TCP_PORT.getPropertyName());
-
-        final List<Server> aliveServers = serverAPI
-                .getAliveServers(Collections.singletonList(currentServer.getServerId()));
-
-        currentServer = Server.builder(currentServer)
-                .withEsTransportTcpPort(Integer.parseInt(transportTCPPort)).build();
-        aliveServers.add(currentServer);
+        final String transportTCPPort = externalSettings.get(ServerPort.ES_TRANSPORT_TCP_PORT.getPropertyName());
 
-        final StringBuilder initialHosts = new StringBuilder();
+        final List<Server> aliveServers = serverAPI.getAliveServers();
 
-        int i=0;
-        for (Server server : aliveServers) {
-            if(i>0) {
-                initialHosts.append(",");
-            }
+        String initialHosts = aliveServers.stream().map(this::getServerAddress).collect(Collectors.joining(","));
```

Comment:
Stream and collectors instead of all the manual comma handling and old-fashion loop. Also using method reference to get the full `ipadress:port` out of the server